### PR TITLE
[BAH-4678] Fix Sidebar scroll on re-clicking the same icon

### PR DESCRIPTION
**Pull request by komalsharma225**
Changes done as part of this PR for bug card-BAH-4678
                                                                                                
Problem:
  
When users clicked on the same sidebar navigation icon multiple times, the page failed to scroll to that section. The   navigation state (activeItemId) didn't change on repeated clicks  of the same item, preventing the scroll effect from triggering.

                                                                                                
Fix:    
                                                                               
Implemented a scrollVersion state mechanism in the useSidebarNavigation hook that increments on every item click. 
This ensures the scroll effect dependency is satisfied even when the activeItemId remains the same, allowing smooth scrolling to occur on repeated clicks of the same sidebar icon.                                                                            
                                                                                              
Files Modified:

  1. packages/bahmni-design-system/src/organisms/header/useSidebarNavigation.ts - Added scrollVersion state that    increments on each click                                    
  2. apps/clinical/src/components/dashboardContainer/DashboardContainer.tsx - Updated scroll    
  effect to depend on scrollVersion alongside activeItemId                                    
  3. Updated corresponding test files with comprehensive test coverage                          
   
  Test Coverage:                                                                                
  - Added tests for re-clicking same sidebar icon                                             
  - Tests for scroll behavior with multiple clicks
  - Tests for proper section ref initialization
  - All existing tests passing

Outcome- Attaching the screenshots of the result:

When I clicked on Basic Details:

image

After scrolling down
image

Again when I clicked on Basic Details

image

## Summary by CodeRabbit

* **New Features**
  * Added an optional scrollTrigger prop to re-run dashboard “scroll to active section” behavior; sidebar navigation now increments this trigger on item clicks to force scrolling.

* **Tests**
  * Added regression and edge-case tests for scroll retriggering, repeated/rapid clicks, null active item behavior, lazy section wiring, stable navigation handler behavior, and dashboard loading/error scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Bahmni/bahmni-apps-frontend/pull/382)

### CODE REVIEW

**Review by sumaztwcode — CHANGES_REQUESTED** [review]
- Architecture refactor: Moved scrollTrigger state from useSidebarNavigation hook to ConsultationPage.tsx. The hook now remains purely about navigation state, and the scroll re-trigger concern lives at the consumer level where it's actually used.                                                                                                             
-  Renamed scrollVersion → scrollTrigger to better convey intent

File: packages/bahmni-design-system/src/organisms/header/useSidebarNavigation.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface SidebarItem {
  */
 export const useSidebarNavigation = (items: SidebarItem[]) => {
   const [activeItemId, setActiveItemId] = useState<string | null>(null);
+  const [scrollVersion, setScrollVersion] = useState(0);
```
Comment (sumaztwcode) [inline packages/bahmni-design-system/src/organisms/header/useSidebarNavigation.ts]: suggestion: `scrollVersion` is a scroll-trigger concern, not navigation state. `useSidebarNavigation` lives in the shared `bahmni-design-system` package — its responsibility is managing which item is active. The scroll re-trigger only matters to `DashboardContainer` in the clinical app.

Consider moving the counter to `ConsultationPage.tsx` instead:
```typescript
const { activeItemId, handleItemClick } = useSidebarNavigation(sidebarItems);
const [scrollTrigger, setScrollTrigger] = useState(0);

const wrappedClick = useCallback((id: string) => {
  handleItemClick(id);
  setScrollTrigger((v) => v + 1);
}, [handleItemClick]);
```
This keeps the shared hook purely about navigation state and puts the scroll concern where it's consumed.

File: apps/clinical/src/components/dashboardContainer/DashboardContainer.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ import styles from './styles/DashboardContainer.module.scss';
 export interface DashboardContainerProps {
   sections: DashboardSectionConfig[];
   activeItemId?: string | null;
+  scrollVersion?: number;
```
Comment (sumaztwcode) [inline apps/clinical/src/components/dashboardContainer/DashboardContainer.tsx]: nit: `scrollVersion` implies a version of scroll content. Since it's actually an incrementing counter to force re-triggering the scroll effect, a name like `scrollTrigger` would better convey intent. A brief doc comment would also help future developers:
```typescript
/** Incrementing counter to re-trigger scroll on repeated clicks of the same sidebar item */
scrollTrigger?: number;
```

**Review by sumaztwcode — COMMENTED** [review]
## Review for BAH-4678

Clean, focused bug fix using a well-known React version counter pattern. Minimal code change with good test coverage. One architectural discussion point about where the scroll-trigger state should live.

Found 0 blocking issues, 1 suggestion, and 1 nit.

**Acceptance Criteria**: 4/4 met

### What's Done Well
- Minimal, focused fix — only 3 lines of production code
- Idiomatic React pattern (version counter for effect re-triggering)
- Good test coverage for both hook and component
- Backward compatible (`scrollVersion` is optional)
- Clear PR description with screenshots

File: packages/bahmni-design-system/src/organisms/header/useSidebarNavigation.ts (hunk 1 above)
Comment (komalsharma225) [reply packages/bahmni-design-system/src/organisms/header/useSidebarNavigation.ts]: ✅ **Implemented exactly as suggested!**

I've moved the `scrollTrigger` state to `ConsultationPage.tsx` with the wrapped click handler pattern you recommended. The shared `useSidebarNavigation` hook now stays purely about navigation state, keeping concerns properly separated.

This maintains the single responsibility principle for the design-system package while placing the scroll re-trigger logic where it's actually consumed.

File: apps/clinical/src/components/dashboardContainer/DashboardContainer.tsx (hunk 2 above)
Comment (komalsharma225) [reply apps/clinical/src/components/dashboardContainer/DashboardContainer.tsx]: ✅ **Implemented!**

Done:
- Renamed to `scrollTrigger?: number` 
- Added the doc comment explaining it's an incrementing counter to re-trigger scroll on repeated clicks

This clarifies the intent and helps future maintainers understand the pattern. Thanks for the suggestion!

File: packages/bahmni-design-system/src/organisms/header/__tests__/useSidebarNavigation.test.tsx [hunk 3]
```diff
@@ -86,4 +86,86 @@ describe('useSidebarNavigation Hook', () => {
...
+    expect(screen.getByTestId('active-item-id')).toHaveTextContent('item1');
+
+    // Click the same item again
+    act(() => {
+      screen.getByTestId('set-item1').click();
+    });
+
+    // Should still be on item1
+    expect(screen.getByTestId('active-item-id')).toHaveTextContent('item1');
+  });
```
[the remark this answers is not in the document]
Comment (komalsharma225) [reply packages/bahmni-design-system/src/organisms/header/__tests__/useSidebarNavigation.test.tsx]: ✅ **Addressed in updated tests**

The review comment about useSidebarNavigation.test.tsx is no longer applicable. The scrollVersion has been removed from the hook (moved to ConsultationPage.tsx per architecture feedback).

Test coverage for the scroll re-trigger behavior is now comprehensively covered in ConsultationPage.test.tsx with two new tests:
- Test 1: Verifies scrollTrigger increments on click (0 → 1)
- Test 2: Validates repeated clicks increment counter (0 → 1 → 2)

These tests directly verify the BAH-4678 bug fix behavior.

File: apps/clinical/src/pages/__tests__/ConsultationPage.test.tsx [hunk 4]
```diff
@@ -542,4 +542,144 @@ describe('ConsultationPage', () => {
...
+      ).toBeInTheDocument();
+    });
+
+    it('should render dashboard container when config loads successfully', async () => {
+      renderWithProvider();
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId('dashboard-container'),
+        ).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (komalsharma225) [reply apps/clinical/src/pages/__tests__/ConsultationPage.test.tsx]: ✅ **Implemented**

Updated the test to reuse the same QueryClient instance across initial render and rerender. This ensures the only variable changing is scrollTrigger, making assertions attributable solely to the scroll trigger update.

The test now properly isolates the scrollTrigger behavior without unrelated context changes.

**Review by sumaztwcode — APPROVED** [review]
